lib/icmp/icmpv4.ml

```diff
@@ -0,0 +1,87 @@
+open Result
+
+module Make(IP : V1_LWT.IPV4) = struct
+  type buffer = Cstruct.t
+  type 'a io = 'a Lwt.t
+  type ipaddr = Ipaddr.V4.t
+
+  type t = {
+    ip : IP.t;
+    echo_reply : bool;
+  }
+
+  let pp_dst_unreachable buf =
+    Printf.sprintf "ICMP Destination Unreachable: %s\n%!" @@
+    match Icmpv4_wire.get_icmpv4_code buf with
+    | 0  -> "Destination network unreachable"
+    | 1  -> "Destination host unreachable"
+    | 2  -> "Destination protocol unreachable"
+    | 3  -> "Destination port unreachable"
+    | 4  -> "Fragmentation required, and DF flag set"
+    | 5  -> "Source route failed"
+    | 6  -> "Destination network unknown"
+    | 7  -> "Destination host unknown"
+    | 8  -> "Source host isolated"
+    | 9  -> "Network administratively prohibited"
+    | 10 -> "Host administratively prohibited"
+    | 11 -> "Network unreachable for TOS"
+    | 12 -> "Host unreachable for TOS"
+    | 13 -> "Communication administratively prohibited"
+    | 14 -> "Host Precedence Violation"
+    | 15 -> "Precedence cutoff in effect"
+    | code -> Printf.sprintf "Unknown ICMP code: %d" code
```

Comment:
(imho an indirection is missing, a cenum from number to icmp type, and the formatter should use the high-level icmp type, rather than numbers... but this can all wait for another PR)